portal/src/graphql/adminapi/UsersList.tsx

```diff
@@ -33,6 +34,41 @@ interface Props {
   onChangeOffset?: (offset: number) => void;
 }
 
+interface UserInfo {
+  username: string;
+  phone: string;
+  email: string;
+}
+
+function extractUserInfoFromIdentities(
```

Comment:
I suppose this function would be used in other pages (e.g. user details). We may want to extract it into another file. Therefore, the input arguments should not depending on a specific query. Instead, we can define a interface structural compatible with the query type, with only the fields required by the function.

Reply:
If we are going to extract it to a util function, the fields should allow undefined so different caller can use their preferred placeholder string. 

```typescript
interface UserInfo {
  username?: string;
  phone?: string;
  email?: string;
}
```

and in this particular component, we should write 

```typescript
const PLACEHOLDER = "-";

<span>{userInfo.username ?? PLACEHOLDER }}</span>
```

Reply:
> we can define a interface structural compatible with the query type

So we need to cast the query result to input type when using the function?

Reply:
TypeScript uses structural typing. Simply speaking, two interface with compatible fields can be assigned to each other without casting. So the `UsersListQuery_users_edges_node_identities` can be accepted into a `UserInfo` type mentioned above.

Reply:
I see, I defined my own compatible interface and there is no warning